### PR TITLE
feat: 1208 endpoint to retrieve dataset identifying information based on explorer_url

### PR DESCRIPTION
**Pull request by MDunitz**
### Reviewers
**Functional:** 
- @maniarathi 
**Readability:** 

---

## Changes
- add endpoint to retrieve dataset identifying information based on explorer url

## Definition of Done (from ticket)
given an explorer_url the endpoint returns the appropriate information

### CODE REVIEW

File: scripts/cxg_admin.py [hunk 1]
```diff
@@ -6,12 +6,14 @@
 import os
 import sys
 
+
```
Comment (maniarathi) [inline scripts/cxg_admin.py:9]: No changes in this file so can delete the line changes?

File: backend/corpora/common/entities/dataset.py [hunk 2]
```diff
@@ -107,6 +108,15 @@ def get(cls, session, dataset_uuid, include_tombstones=False):
                 return None
         return dataset
 
+    @classmethod
+    def get_by_explorer_url(cls, session, explorer_url):
+        """
+        Return the most recently created dataset with the given explorer_url or None
+        """
+        filters = [cls.table.explorer_url == explorer_url]
+        dataset = session.query(cls.table).filter(*filters).order_by(cls.table.created_at.desc()).limit(1).all()
```
[the remark this answers is not in the document]
Comment (MDunitz) [reply backend/corpora/common/entities/dataset.py:117]: Discussed further via zoom. Determined we will differentiate between published and revised datasets (when viewed in explorer) by appending /revised to the end of the explorer_url, removing the suffix when publishing the dataset

File: tests/unit/backend/chalice/api_server/test_v1_dataset.py [hunk 3]
```diff
@@ -228,6 +234,89 @@ def test__cancel_dataset_download__user_not_logged_in(self):
...
+                "dataset_id": dataset_without_artifacts.id,
+                "collection_id": public_collection.id,
+                "tombstoned": False,
+            }
+
+            response = self.app.get(test_url_no_cxg_artifact, headers)
+            response.raise_for_status()
+            self.assertEqual(json.loads(response.body), expected_identifiers)
+
+    def test__dataset_meta__404(self):
```
[the remark this answers is not in the document]
Comment (MDunitz) [reply tests/unit/backend/chalice/api_server/test_v1_dataset.py:312]: Removed the possible 403 response. We arent checking for auth on this route

**Review by maniarathi — APPROVED** [review]
LGTM!

**Pull request merged**